parts/linux/cloud-init/artifacts/aks-diagnostic.py

```diff
@@ -0,0 +1,310 @@
+#! /usr/bin/env python3
+
+"""
+aks-diagnostic.py
+
+This script uploads a diagnostic log file to an Azure Blob Storage container using a REST API call.
+Usage:
+    python aks-diagnostic.py <container> <storage_account_url> <sas_token> [file_path]
+
+Arguments:
+    container              : Name of the blob container.
+    storage_account_url    : Full URL of the Azure storage account (e.g., https://<account>.blob.core.windows.net/).
+    sas_token              : Shared Access Signature token for authentication.
+    file_path (optional)   : Path to the log file to upload (default: /var/lib/waagent/logcollector/logs.zip).
+"""
+import requests
```

Comment:
I don't think it's part of the standard library. Are you sure you will have it no the VM?

Reply:
Additionally, with python you have risk of the breaking once python version is updated.
Python doesn't have same backwards compatibility gurantee as golang.